Plugin submodules escape 'go test ./...' coverage; needs aggregator

## Problem

Running `go test ./...` (or `go test -short ./...`) from the repo root tests only the root module. The three plugin submodules — `plugins/memory`, `plugins/postgres`, `plugins/sqlite` — each have their own `go.mod`, and Go's `./...` recursion does NOT cross module boundaries. `go.work` doesn't save you: `go test` still evaluates `./...` relative to the current module.

This has already caused one escape: during PR #44's end-of-deliverable verification, `go test -short ./...` passed from the root while 18 sqlite plugin tests were silently broken by the `defer m.Close()` regression in `checkSchemaCompat`. The bug survived until PR #45's pre-merge review caught it. See PR #45 commit `fc8a6f1` for the fix.

## Evidence

From the feat/provisioning-desktop-layer branch before the fix:

```
$ go test -short ./... | grep -cE "^(ok|FAIL)"
40  (all passing)

$ cd plugins/sqlite && go test -short ./...
--- FAIL: TestSearcher_ORFilter ...
--- FAIL: TestSearcher_PostFilterRegex ...
--- FAIL: TestNewStoreFactoryForTest ...
... 18 failures total
```

## Existing partial coverage

`.github/workflows/ci.yml` has a `per-module-hygiene` job that runs `GOWORK=off go build ./...` and `GOWORK=off go vet ./...` against each plugin directory. It does NOT run plugin tests. Build passing does not catch runtime regressions.

## Suggested fixes

Prefer (a) and (b); either is sufficient, both is nicer.

**(a) Makefile `test-all` target.** Aggregates root + all plugins into one invocation. Documented in `CLAUDE.md`'s Common Commands as the canonical "test everything" command.

```makefile
test-all:
	go test -short ./...
	cd plugins/memory && go test -short ./...
	cd plugins/postgres && go test -short ./...  # requires Docker
	cd plugins/sqlite && go test -short ./...
```

**(b) CI extension.** Extend `per-module-hygiene` (or add a sibling job `per-module-tests`) to run `go test -short ./...` inside each plugin directory. The postgres tests need Docker; testcontainers-go handles this on `ubuntu-latest`.

**(c) CLAUDE.md clarification.** The Common Commands block currently says `Test (all): go test ./... -v (includes E2E — requires Docker)`. Update to name what's covered (root module + internal E2E) and point at the `make test-all` target once it exists.

## Acceptance criteria

- [ ] A single local command runs root tests and all three plugin-submodule test suites.
- [ ] CI runs plugin-submodule tests on every push.
- [ ] CLAUDE.md accurately describes which command tests what.

## Out of scope

- Full E2E / integration suite changes (covered separately).
- Race detector policy (separate; already documented).

## Observed by

PR #45 code review (https://github.com/Cyoda-platform/cyoda-go/pull/45), fix commit `fc8a6f1`.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -23,6 +23,22 @@ jobs:
       - name: Run tests with race detector
         run: go test -race ./...
 
+      # Plugin submodules each have their own go.mod, so `./...` from the root
+      # does not recurse into them. Run tests inside each module so CI does
+      # not miss regressions (issue #46). Postgres plugin uses testcontainers-go;
+      # ubuntu-latest provides the Docker socket.
+      - name: Run plugins/memory tests
+        working-directory: plugins/memory
+        run: go test -v ./...
+
+      - name: Run plugins/sqlite tests
+        working-directory: plugins/sqlite
+        run: go test -v ./...
+
+      - name: Run plugins/postgres tests
+        working-directory: plugins/postgres
+        run: go test -v ./...
+
       - name: Build
         run: go build -o bin/cyoda ./cmd/cyoda
 

--- a/CLAUDE.md
+++ b/CLAUDE.md
@@ -107,8 +107,8 @@ aggregator targets below when you want coverage across the whole repo.
 - Test (root + every plugin submodule): `make test-all` — covers root + `plugins/memory|sqlite|postgres`; Docker required for postgres testcontainers
 - Test (root + plugins, short): `make test-short-all`
 - Test (E2E only): `go test ./internal/e2e/... -v`
-- Coverage: `go test -coverprofile=coverage.out ./...`
-- Race detector: `go test -race ./...`
+- Coverage (root module): `go test -coverprofile=coverage.out ./...` — run inside each `plugins/*` for per-plugin coverage
+- Race detector (root module): `go test -race ./...`
 - Build: `go build -o bin/cyoda ./cmd/cyoda`
 - Tidy: `go mod tidy`
-- Vet: `go vet ./...`
+- Vet (root module): `go vet ./...` — the `per-module-hygiene` CI job vets each plugin separately

--- a/CLAUDE.md
+++ b/CLAUDE.md
@@ -97,8 +97,15 @@ Review and security audit prevent defects reaching main.
 
 ## Common Commands
 
-- Test (all): `go test ./... -v` (includes E2E — requires Docker)
-- Test (unit only): `go test -short ./... -v`
+Plugin submodules (`plugins/memory`, `plugins/sqlite`, `plugins/postgres`)
+each have their own `go.mod`; Go's `./...` recursion does **not** cross
+module boundaries, so root-module commands miss them. Use the `-all`
+aggregator targets below when you want coverage across the whole repo.
+
+- Test (root module): `go test ./... -v` — root incl. internal/e2e (requires Docker)
+- Test (root module, unit only): `go test -short ./... -v`
+- Test (root + every plugin submodule): `make test-all` — covers root + `plugins/memory|sqlite|postgres`; Docker required for postgres testcontainers
+- Test (root + plugins, short): `make test-short-all`
 - Test (E2E only): `go test ./internal/e2e/... -v`
 - Coverage: `go test -coverprofile=coverage.out ./...`
 - Race detector: `go test -race ./...`

--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,9 @@
-.PHONY: dev-up dev-down dev-ps dev-logs dev-run dev-test build test clean docker-build docker-push todos
+.PHONY: dev-up dev-down dev-ps dev-logs dev-run dev-test build test test-all test-short-all clean docker-build docker-push todos
+
+# Plugin submodules: each has its own go.mod, so `go test ./...` from the
+# repo root does not recurse into them. The aggregator targets below close
+# that coverage gap (issue #46).
+PLUGIN_MODULES := plugins/memory plugins/sqlite plugins/postgres
 
 # --- Docker services ---
 
@@ -48,8 +53,22 @@ endif
 
 # --- Testing ---
 
-test:                  ## Run all tests (postgres tests skipped without DB)
+test:                  ## Run root-module tests only (plugin submodules skipped — see test-all)
 	go test ./... -v
+
+test-all:              ## Run root + every plugin submodule (requires Docker for postgres)
+	go test ./... -v
+	@for m in $(PLUGIN_MODULES); do \
+	  echo "==> go test ./... in $$m"; \
+	  (cd $$m && go test ./... -v) || exit $$?; \
+	done
+
+test-short-all:        ## Run root + every plugin submodule with -short (quick coverage check)
+	go test -short ./... -v
+	@for m in $(PLUGIN_MODULES); do \
+	  echo "==> go test -short ./... in $$m"; \
+	  (cd $$m && go test -short ./... -v) || exit $$?; \
+	done
 
 dev-test: dev-up       ## Run all tests against local postgres
 	set -a && . .env.dev && set +a && go test ./... -v -count=1

